File: Contracts/Model/Product/Response/ProductDetailResponse.cs
﻿namespace Contracts.Model.Product.Response;

public class ProductDetailResponse : BaseResponse
{
	public Guid Id { get; set; }
	public int? CategoryId { get; set; }
	public Guid? OptionId { get; set; }
	public bool IsService { get; set; }
	public WarrantyPreviewResponse Warranty { get; set; }
	public OptionDetailResponse Option { get; set; }
	public ICollection<OptionPreviewResponse> Options { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/Response/productDetailResponse.ts
import {BaseResponse} from "../../baseResponse";
import {OptionDetailResponse} from "./optionDetailResponse";
import {OptionPreviewResponse} from "./optionPreviewResponse";
import {WarrantyResponse} from "@core/interfaces/Product/Response/warrantyResponse";

export interface ProductDetailResponse extends BaseResponse {
  id: string;
  isService: boolean;
  categoryId: number | null;
  optionId: string | null;
  warranty: WarrantyResponse;
  option: OptionDetailResponse;
  options: OptionPreviewResponse[];
}
